### PR TITLE
awscli2: init at 2.0.26

### PR DESCRIPTION
**Pull request by davegallant**
#### Motivation for this change

"AWS CLI v2 supports configuring and using credentials through AWS SSO using the new aws configure sso command."
https://aws.amazon.com/blogs/developer/aws-cli-v2-is-now-generally-available/

Since AWS CLI version 1 will be supported for a while, it may make sense to keep this a separate derivation.

https://github.com/NixOS/nixpkgs/issues/80956

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/awscli2/default.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+in with py.pkgs; buildPythonApplication rec {
+  pname = "awscli2";
+  version = "2.0.26"; # N.B: if you change this, change botocore to a matching version too
+
+  src = fetchFromGitHub {
+      owner = "aws";
+      repo = "aws-cli";
+      rev = version;
+      sha256 = "sha256:1ysmr17gbcj6vs9ywzwgvd9caxwxgg9bnfvvkyks4fii34ji5qq8";
```
Comment (bhipple) [inline pkgs/tools/admin/awscli2/default.nix:38]:
```suggestion
      hash = "sha256:1ysmr17gbcj6vs9ywzwgvd9caxwxgg9bnfvvkyks4fii34ji5qq8";
```

File: pkgs/tools/admin/awscli2/default.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    };
```
Comment (bhipple) [inline pkgs/tools/admin/awscli2/default.nix:39]: Indentation is messed up here; give it a run through https://github.com/nix-community/nixpkgs-fmt to fix automatically.

**Review by bhipple — CHANGES_REQUESTED** [review]
Thanks for taking this on!

File: pkgs/tools/admin/awscli2/default.nix (hunk 2 above)
Comment (davegallant) [reply pkgs/tools/admin/awscli2/default.nix]: Thanks for the feedback. I applied the suggestions.

**Comment by bhipple** [comment]
LGTM; can you fixup git history per https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md ?

**Comment by davegallant** [comment]
> LGTM; can you fixup git history per https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md ?

@bhipple Fixed :wrench:

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 91655` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- awscli2
</details>

**Review by bhipple — APPROVED** [review]

**Pull request merged**